src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,157 @@
+        int y = size;
+        T[] newArray;
+        if (sizesSum > internalArray.length) {
+            newArray = (T[]) new Object[sizesSum];
+            for (int i = 0; i < size; i++) {
+                newArray[i] = internalArray[i];
+            }
+            for (int i = 0; i < list.size(); i++, y++) {
+                newArray[y] = list.get(i);
+            }
+            internalArray = newArray;
+        } else {
+            for (int i = 0; i < list.size(); i++, y++) {
+                internalArray[y] = list.get(i);
+            }
+            size = sizesSum;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return internalArray[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= 0 && index < size) {
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T temporaryElement;
+        if (index >= 0 && index < size) {
+            temporaryElement = internalArray[index];
+            for (int i = index; i < size - 1; i++) {
+                internalArray[i] = internalArray[i + 1];
+            }
+            size--;
+            return temporaryElement;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound array!");
+        }
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = searchIndex(element);
+        if (index >= 0 && index < size) {
+            return remove(index);
+        } else {
+            throw new NoSuchElementException("Index is out of bound array!");
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        if (size == 0) {
+            return true;
+        }
         return false;
     }
+
+    public void grow() {
+        int newLength = (int) (internalArray.length + DEFAULT_MULTIPLIER);
+        T[] newArray = (T[]) new Object[newLength];
+        for (int i = 0; i < size; i++) {
+            newArray[i] = internalArray[i];
+        }
+        internalArray = newArray;
+    }
+
+    public int searchIndex(T element) {
+        if (element == null) {
+            for (int i = 0; i < size; i++) {
+                if (internalArray[i] == null) {
+                    return i;
+                }
+            }
+        } else if (element != null) {
+            for (int i = 0; i < size; i++) {
+                if (element.equals(internalArray[i])) {
+                    return i;
+                }
+            }
+        } else {
```

Comment:
lets remove the outer if else, and combine 

if (internalArray[i] == null) 
and
if (element.equals(internalArray[i])) {

because it has the same body